components/raftstore/src/store/fsm/store.rs

```diff
@@ -258,7 +259,11 @@ pub struct PollContext<T, C: 'static> {
     pub ready_res: Vec<(Ready, InvokeContext)>,
     pub need_flush_trans: bool,
     pub queued_snapshot: HashSet<u64>,
-    pub current_time: Option<Timespec>,
+    /// `Instant` shared by FSMs within same batch, the underlying value will be seted at
+    /// its first use and will be cleared at `PollHandler::end`. For the same FSM, this
+    /// `Instant` will be monotonic, because a FSM will not be handled in diffrent batch
+    /// at the same time.
+    pub current_time: RefCell<SharedInstant>,
```

Comment:
Why make it `RefCell`?

Reply:
`SharedInstant::get` take `&mut self`, in some situation we need to call `SharedInstant::get` while `StoreMeta` guard in holding, which break the borrow rules.

Reply:
Why not change it to `&mut`? Besides, what's the point of `SharedInstant`? It doesn't seem to introduce any value.

Reply:
Do you means `current_time: &mut SharedInstant` ? And I have replace  `SharedInstant` with `PollContext::get_current_time` PTAL.

Reply:
I mean change `&PollContext` to `&mut PollContext`.

Reply:
> `SharedInstant::get` take `&mut self`, in some situation we need to call `SharedInstant::get` while `StoreMeta` guard in holding, which break the borrow rules.

The problem is we can't have `&self` (holding `store_meta` guard) and `&mut self` at the same time, changing `&PollContext` to `&mut PollContext` didn't help.